process/subprocesses/ghsci.py

```diff
@@ -737,8 +733,6 @@ def _region_data_setup(
                 region = region_config['codename']
             else:
                 region = self.codename
-            if data not in region_config:

```

Comment:
Why remove this?  Just not sure how it relates to earth engine integration.  The logic is admittedly complex, but it is well tested by now.  It can be hard to know how what might seem like a simple change could cascade and cause problems.  So, just checking if there is a reason for this refactoring here, and if not, I'd be inclined to leave it as beyond scope.  It could involve unintended consequences, potentially.